src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -114,6 +114,10 @@ public Person toModelType() throws IllegalValueException {
         if (address != null && !Address.isValidAddress(address)) {
             throw new IllegalValueException(Address.MESSAGE_CONSTRAINTS);
         }
+
+        if (address == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Address.class.getSimpleName()));
+        }
```

Comment:
This line needs explanation.

Reply:
this was part of the JsonAdaptedPerson this file got added in this branch so it was failing a JsonAdaptedPersonTest, had to fix this to make the test pass.

Reply:
But non-professors need not have an address? Have you tried storing a non-professor into the data file, exit the program and reopen it?

Reply:
i'm not sure what you mean exactly. the test file was failing because with a null address and exception was expected to be thrown but it wasn't so i just added this in.

Reply:
I refuse to merge. This issue isn't addressed sufficiently early.